packages/SwingSet/src/lib/id.js

```diff
@@ -2,6 +2,9 @@ import { Nat } from '@endo/nat';
 
 import { Fail } from '@agoric/assert';
 
+const vatIDPattern = /^v\d+$/;
+const deviceIDPattern = /^d\d+$/;
```

Comment:
I still don't think we want to allow e.g. "v0001", which would never be produced by `makeVatID`.
```suggestion
const vatIDPattern = /^v0$|^v[1-9]\d*$/;
const deviceIDPattern = /^d0$|^d[1-9]\d*$/;
```

Reply:
'v0001' are accepted IDs by old implementation as well. We can add more code in insistVatID to check for them but the benefit of that extra check might not cover the cost of extra complexity. My suggestion would be to move forward with current change that fixes `v` and improves the performance with overall simpler code.

Reply:
Fair point, and I don't think it blocks merge. But I would like @warner's opinion on whether or not we should add that strictness (e.g. in a followup).

Reply:
leading `0` being allowed is really a concern for me. It's fine if we don't ever round trip ids to index and back, but if someday we ever start doing so, then the fact that there exists more than 1 id for a single index is a problem.